ansible_runner/utils/capacity.py

```diff
@@ -22,16 +27,27 @@ def get_mem_in_bytes():
 
 
 def get_uuid():
-    try:
-        with open('/var/lib/dbus/machine-id') as f:
-            uuid = f.read()
-        return uuid.strip()
-    except FileNotFoundError:
-        try:
-            with open('/etc/machine-id') as f:
-                uuid = f.read()
-                return uuid.strip()
-        except FileNotFoundError:
-            error = ("Could not find /var/lib/dbus/machine-id or "
-                     "/etc/machine-id files, UUID undiscoverable.")
-            return error
+    uuid_directory = Path('/etc/ansible/facts.d')
+    if uuid_directory.exists():
+        # Read the contents of the uuid.txt file if it already exists
+        uuid_file_path = os.path.join(uuid_directory[0], 'uuid.txt')
+        with open(uuid_file_path) as f:
+            saved_uuid = f.read()
+        return saved_uuid
+    else:
+        # Generate a new UUID if no uuid.txt file is found
+        newly_generated_uuid = generate_uuid()
+        return newly_generated_uuid
+
+
+def generate_uuid():
+    generated_uuid = str(uuid.uuid4())
+
+    # Store the newly-generated UUID in a new dir/file
+    path = '/etc/ansible/facts.d'
+    uuid_file = 'uuid.txt'
+    uuid_file_path = pathlib.Path(path) / uuid_file
```

Comment:
Is there any reason why this can't just be `os.path.join(path, uuid_file)`?

Reply:
That would change the result from a `Path` object to a `str`.